### PR TITLE
Feature/witi 217 create custom fields entity on cost centre md table

### PR DESCRIPTION
**Pull request by RamanSel**
#### What problem is this solving?

Adding custom fields to organizations. [WITI-217](https://syatt.atlassian.net/browse/WITI-217).

#### How to test it?

[Workspace](https://witi217--whitebird.myvtex.com/admin/b2b-organizations/organizations/#/settings)

#### Screenshots or example usage:

witi217--whitebird myvtex com_admin_b2b-organizations_organizations_ (1)

#### Related to / Depends on

Depends on [b2b-organizations-graphql part](https://github.com/syatt-io/b2b-organizations-graphql/pull/2).

### CODE REVIEW

**Review by jonmast — CHANGES_REQUESTED** [review]

File: react/admin/AutoApproveSettings.tsx [hunk 1]
```diff
@@ -1,33 +1,40 @@
-import React, { useEffect, useState, useContext } from 'react'
```
[the remark this answers is not in the document]
Comment (RamanSel) [reply react/admin/AutoApproveSettings.tsx]: ✅ moved component to a separate file and route

File: react/admin/AutoApproveSettings.tsx [hunk 2]
```diff
@@ -1,33 +1,40 @@
...
 
 export interface CellRendererProps<RowType> {
   cellData: unknown
   rowData: RowType
   updateCellMeasurements: () => void
 }
 
+export interface CustomField {
+  name: string
+  type: string
```
[the remark this answers is not in the document]
Comment (RamanSel) [reply react/admin/AutoApproveSettings.tsx]: ✅ Added type: 'text' for FE and BE

File: react/admin/OrganizationCustomFields.tsx [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+  handleUpdate: (index: number, customField: CustomField) => void
+}
+
+const CustomField: React.FC<CustomFieldProps> = ({
+  index,
+  handleUpdate,
+  customFieldState,
+}) => {
+  const { formatMessage } = useIntl()
+  const [customField, setCustomField] = useState<CustomField>({
```
[the remark this answers is not in the document]
Comment (RamanSel) [reply react/admin/OrganizationCustomFields.tsx]: ✅ state now lives inside 'OrganizationCustomFields' component

**Comment by RamanSel** [comment]
@jonmast, 
I've prettified the code a bit by using their other files as an example, please advice if it's good enough to submit a PR.

File: react/admin/OrganizationCustomFields.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (jonmast) [reply react/admin/OrganizationCustomFields.tsx]: I'm still seeing a state variable in the field component here: https://github.com/syatt-io/b2b-organizations/pull/2/files#diff-e867d185ea29743c679c5578bc3d1ec02be6aa5f97b264114b3b9b8c687ede14R24

**Comment by jonmast** [comment]
@RamanSel You'll want to make sure `yarn lint` isn't returning any errors before submitting, that would cause CI to fail

**Review by jonmast — CHANGES_REQUESTED** [review]
I'm still seeing the state issue, feel free to ping me if it doesn't make sense and we can discuss further.

**Comment by RamanSel** [comment]
New PR will be submitted that includes logic for saving custom fields input values

**Pull request closed without merging**